components/raftstore/src/store/worker/split_check.rs

```diff
@@ -198,10 +197,35 @@ where
             "end_key" => log_wrappers::Value::key(&end_key),
         );
         CHECK_SPILT_COUNTER.all.inc();
-
+        let bucket_check_policy = policy;
         let mut host =
             self.coprocessor
                 .new_split_checker_host(region, &self.engine, auto_split, policy);
+        if bucket_check_policy == CheckPolicy::Approximate && host.enable_region_bucket() {
```

Comment:
You're right. Maybe host.policy() is better, though it does not work well either in big region with current code because only region_size >= host.cfg.region_max_size.0 will trigger CheckPolicy::approximate. Probably we need to change the code there too to enforce approximate when region size is large enough .

Reply:
should use the host.policy(), and change the condition `region_size >= host.cfg.region_max_size.0` when region size is enlarged.